lib/kafka/connection.rb

```diff
@@ -72,9 +72,10 @@ def close
     #
     # @param request [#encode, #response_class] the request that should be
     #   encoded and written.
+    # @param async if true, do not wait for the response
     #
     # @return [Object] the response.
-    def send_request(request)
+    def send_request(request, async: false)
```

Comment:
Maybe just split out a separate `send_async_request` that is called by `send_request`, after which it would `wait_for_response`.